app/src/main/java/org/simple/clinic/drugs/selectionv2/PrescribedDrugsScreenControllerV2.kt

```diff
@@ -60,25 +60,22 @@ class PrescribedDrugsScreenControllerV2 @Inject constructor(
         .map { (protocolDrugs, prescribedDrugs) ->
 
           val prescribedProtocolDrugs = prescribedDrugs.filter { it.isProtocolDrug }
-          // Select protocol drugs if prescriptions exist for them.
+          val isLast = prescribedDrugs.any { it.isProtocolDrug.not() }
+          // Show dosage if prescriptions exist for them.
           val protocolDrugSelectionItems = protocolDrugs
               .mapIndexed { index: Int, drugAndDosages: ProtocolDrugAndDosages ->
-                val matchingPrescribedDrug =
-                    prescribedProtocolDrugs
-                        .firstOrNull { it.name == drugAndDosages.drugName }
-
+                val matchingPrescribedDrug = prescribedProtocolDrugs.firstOrNull { it.name == drugAndDosages.drugName }
                 ProtocolDrugListItem(
                     id = index,
                     drugName = drugAndDosages.drugName,
-                    prescribedDrug = matchingPrescribedDrug)
+                    prescribedDrug = matchingPrescribedDrug,
+                    hideDivider = isLast.not() && index == protocolDrugs.size - 1)
               }
 
           val customPrescribedDrugItems = prescribedDrugs
               .filter { it.isProtocolDrug.not() }
               .sortedBy { it.updatedAt.toEpochMilli() }
-              .mapIndexed { index: Int, drugAndDosages: PrescribedDrug ->
-                ProtocolDrugListItem(index + protocolDrugSelectionItems.size, drugAndDosages.name, drugAndDosages)
-              }
+              .map { drugAndDosages -> CustomPrescribedDrugListItem(drugAndDosages) }
```

Comment:
Since a prescribed drug has only one drug and dosage, can we rename `drugAndDosages` -> `drugAndDosage` or even just `prescribedDrug` instead?